goosebit/updater/controller/v1/routes.py

```diff
@@ -106,77 +98,66 @@ async def deployment_base(
-
-            reported_firmware = await Firmware.get_or_none(id=data["id"])
-
-            # From hawkBit docu: DDI defines also a status NONE which will not be interpreted by the update server
-            # and handled like SUCCESS.
-            if state == "success" or state == "none":
-                await updater.update_device_state(UpdateStateEnum.FINISHED)
-
-                # not guaranteed to be the correct rollout - see next comment.
-                rollout = await updater.get_rollout()
-                if rollout:
-                    if rollout.firmware == reported_firmware:
-                        rollout.success_count += 1
-                        await rollout.save()
-                    else:
-                        logging.warning(
-                            f"Updating rollout success stats failed, firmware={reported_firmware.id}, device={updater.dev_id}"  # noqa: E501
-                        )
-
-                # setting the currently installed version based on the current assigned firmware / existing rollouts
-                # is problematic. Better to assign custom action_id for each update (rollout id? firmware id? new id?).
-                # Alternatively - but requires customization on the gateway side - use version reported by the gateway.
-                await updater.update_fw_version(reported_firmware.version)
-                logger.debug(f"Installation successful, firmware={reported_firmware.version}, device={updater.dev_id}")
-
-            elif state == "failure":
-                await updater.update_device_state(UpdateStateEnum.ERROR)
-
-                # not guaranteed to be the correct rollout - see comment above.
-                rollout = await updater.get_rollout()
-                if rollout:
-                    if rollout.firmware == reported_firmware:
-                        rollout.failure_count += 1
-                        await rollout.save()
-                    else:
-                        logging.warning(
-                            f"Updating rollout failure stats failed, firmware={reported_firmware.id}, device={updater.dev_id}"  # noqa: E501
-                        )
-
-                logger.debug(f"Installation failed, firmware={reported_firmware.version}, device={updater.dev_id}")
-
-    except KeyError as e:
-        logging.warning(f"Processing deployment feedback failed, error={e}, device={updater.dev_id}")
+    if data.status.execution == FeedbackStatusExecutionState.PROCEEDING:
+        await updater.update_device_state(UpdateStateEnum.RUNNING)
+        logger.debug(f"Installation in progress, device={updater.dev_id}")
+
+    elif data.status.execution in [
+        FeedbackStatusExecutionState.CLOSED,
+        FeedbackStatusExecutionState.CANCELED,
+        FeedbackStatusExecutionState.REJECTED,
+    ]:
+        await updater.update_force_update(False)
+        await updater.update_log_complete(True)
+
+        # TODO: check this, not in DDI schema
```

Comment:
This is not mentioned anywhere in the hawkbit DDI docs, but I added this just in case.

Reply:
DDI just talks about an abstract action_id. gooseBit could create for each update a new record and use its id to track the update. To prevent huge tables (and slow processing) I decided to use the firmware_id as the action_id. This allows to precisely track which firmware got installed.

Reply:
Oh, the ID isn't getting passed through the API request, it's getting passed as a variable in the URL?  That makes a lot more sense, I forgot that change was made.  I'll remove this and make sure that's referenced properly.